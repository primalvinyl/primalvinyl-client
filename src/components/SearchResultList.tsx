import React from 'react';
import { useTable } from 'react-table';
import { Link } from 'react-router-dom';
import styles from './SearchResultList.module.scss';

const Table = ({ columns, data }: any) => {
    const {
        getTableProps,
        getTableBodyProps,
        headerGroups,
        prepareRow,
        rows
    } = useTable({
        columns,
        data
    });
    
    return (
        <table {...getTableProps()}>
            <thead>
                {headerGroups.map(headerGroup => (
                    <tr {...headerGroup.getHeaderGroupProps()}>
                        {headerGroup.headers.map(column => (
                            <th {...column.getHeaderProps()}>
                                {column.render('Header')}
                            </th>
                        ))}
                    </tr>
                ))}
            </thead>
            <tbody {...getTableBodyProps()}>
                {rows.map(row => {
                    prepareRow(row)
                    return (
                        <tr {...row.getRowProps()}>
                            {row.cells.map(cell => (
                                <td {...cell.getCellProps()}>
                                    {cell.render('Cell')}
                                </td>
                            ))}
                        </tr>
                    )
                })}
            </tbody>
        </table>
    )
}

const SearchResultList = ({ list }: SearchResultsProps): React.ReactElement => {
    const columns = React.useMemo(() => [
        {
            Header: '',
            Cell: (table: any) => <img src={table.value} alt="" />,
            accessor: 'song_image_url',
        },
        {
            Header: 'Song',
            Cell: (table: any) => <Link to={`/songs/${table.row.original.id}`}>{table.value}</Link>,
            accessor: 'song_title',
        },
        {
            Header: 'Artist',
            accessor: 'artist_name',
        },
    ], []);

    return (
        <div className={styles.root}>
            <Table columns={columns} data={list} />
        </div>
    )
};

type SearchResultsProps = {
    readonly list: any;
}

export default SearchResultList;
