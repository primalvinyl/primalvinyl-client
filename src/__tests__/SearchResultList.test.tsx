import React from 'react';
import { shallow } from 'enzyme';
import Renderer from 'react-test-renderer';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import rootReducer from '../store/reducers';
import SearchResultList from '../components/SearchResultList';

describe('SearchResultList', () => {
    const store = createStore(rootReducer, {});
    const searchList = [1, 2, 3];

    it('renders', () => {
        const myWrapper = shallow(
            <Provider store={store}>
                <MemoryRouter>
                    <SearchResultList list={searchList} />
                </MemoryRouter>
            </Provider>
        );
        expect(myWrapper.exists());
    });
    
    it('matches snapshot', () => {
        const component = Renderer.create(
            <Provider store={store}>
                <MemoryRouter>
                    <SearchResultList list={searchList} />
                </MemoryRouter>
            </Provider>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});
