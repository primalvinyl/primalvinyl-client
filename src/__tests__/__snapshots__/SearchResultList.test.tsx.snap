// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultList matches snapshot 1`] = `
<div
  className="root"
>
  <table
    role="table"
  >
    <thead>
      <tr
        role="row"
      >
        <th
          colSpan={1}
          role="columnheader"
        >
          
        </th>
        <th
          colSpan={1}
          role="columnheader"
        >
          Song
        </th>
        <th
          colSpan={1}
          role="columnheader"
        >
          Artist
        </th>
      </tr>
    </thead>
    <tbody
      role="rowgroup"
    >
      <tr
        role="row"
      >
        <td
          role="cell"
        >
          <img
            alt=""
          />
        </td>
        <td
          role="cell"
        >
          <a
            href="/songs/undefined"
            onClick={[Function]}
          />
        </td>
        <td
          role="cell"
        >
          
        </td>
      </tr>
      <tr
        role="row"
      >
        <td
          role="cell"
        >
          <img
            alt=""
          />
        </td>
        <td
          role="cell"
        >
          <a
            href="/songs/undefined"
            onClick={[Function]}
          />
        </td>
        <td
          role="cell"
        >
          
        </td>
      </tr>
      <tr
        role="row"
      >
        <td
          role="cell"
        >
          <img
            alt=""
          />
        </td>
        <td
          role="cell"
        >
          <a
            href="/songs/undefined"
            onClick={[Function]}
          />
        </td>
        <td
          role="cell"
        >
          
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
